extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -417,28 +417,88 @@ public long getViewCount() throws ParsingException {
+            final JsonArray topLevelButtons = getVideoPrimaryInfoRenderer()
                     .getObject("videoActions")
                     .getObject("menuRenderer")
-                    .getArray("topLevelButtons")
-                    .getObject(0)
-                    .getObject("toggleButtonRenderer")
-                    .getObject("defaultText")
-                    .getObject("accessibility")
+                    .getArray("topLevelButtons");
+
+            // Try first with the new video actions buttons data structure
+            JsonObject likeToggleButtonRenderer = topLevelButtons.stream()
+                    .filter(JsonObject.class::isInstance)
+                    .map(JsonObject.class::cast)
+                    .map(button -> button.getObject("segmentedLikeDislikeButtonRenderer")
+                            .getObject("likeButton")
+                            .getObject("toggleButtonRenderer"))
+                    .filter(toggleButtonRenderer -> !isNullOrEmpty(toggleButtonRenderer))
+                    .findFirst()
+                    .orElse(null);
+
+            // Use the old video actions buttons data structure if the new one isn't returned
+            if (likeToggleButtonRenderer == null) {
+                /*
+                In the old video actions buttons data structure, there are 3 ways to detect whether
+                a button is the like button, using its toggleButtonRenderer:
+                - checking whether toggleButtonRenderer.targetId is equal to watch-like;
+                - checking whether toggleButtonRenderer.defaultIcon.iconType is equal to LIKE;
+                - checking whether
+                  toggleButtonRenderer.toggleButtonSupportedData.toggleButtonIdData.id
+                  is equal to TOGGLE_BUTTON_ID_TYPE_LIKE.
+                */
+                likeToggleButtonRenderer = topLevelButtons.stream()
+                        .filter(JsonObject.class::isInstance)
+                        .map(JsonObject.class::cast)
+                        .map(topLevelButton -> topLevelButton.getObject("toggleButtonRenderer"))
+                        .filter(toggleButtonRenderer -> toggleButtonRenderer.getString("targetId")
+                                .equalsIgnoreCase("watch-like")
+                                || toggleButtonRenderer.getObject("defaultIcon")
+                                .getString("iconType")
+                                .equalsIgnoreCase("LIKE")
+                                || toggleButtonRenderer.getObject("toggleButtonSupportedData")
+                                .getObject("toggleButtonIdData")
+                                .getString("id")
+                                .equalsIgnoreCase("TOGGLE_BUTTON_ID_TYPE_LIKE"))
+                        .findFirst()
+                        .orElseThrow(() -> new ParsingException(
+                                "Ratings are enabled even though the like button is missing"));
```

Comment:
This is the wrong way around:
```suggestion
                                "The like button is missing even though ratings are enabled"));
```

Reply:
It was already as it is before my changes, but I will change this.

<em>Why I am blamed for code that I didn't wrote :/</em>

Reply:
No one is blaming you. xD We're just striking the iron while it's hot.